src/context/tokenSelector.context.tsx

```diff
@@ -66,6 +73,7 @@ export const TokenContextProvider = ({ children }: { children: React.ReactNode }
         async function fetchAndSetTokenPrice(tokenAddress: string, chainId: string) {
```

Comment:
thought: seems weird to have ITokenData but then in e.g. `fetchAndSetTokenPrice` and other places use different formats (e.g. just tokenAddress and chainid)

Reply:
Hmmm, yes I changes minimally what I needed but there are some things that could be refactored.